opentelemetry-api/src/opentelemetry/trace/propagation/textmap.py

```diff
@@ -20,7 +20,68 @@
 TextMapPropagatorT = typing.TypeVar("TextMapPropagatorT")
 
 Setter = typing.Callable[[TextMapPropagatorT, str, str], None]
-Getter = typing.Callable[[TextMapPropagatorT, str], typing.List[str]]
+
+
+class Getter(typing.Generic[TextMapPropagatorT]):
+    """This class implements a Getter that enables extracting propagated
+    fields from a carrier
+
+    """
+
+    def get(self, carrier: TextMapPropagatorT, key: str) -> typing.List[str]:
+        """Function that can retrieve zero
+        or more values from the carrier. In the case that
+        the value does not exist, returns an empty list.
+
+        Args: carrier: and object which contains values that are used to
+        construct a Context. This object must be paired with an appropriate
+        get_from_carrier which understands how to extract a value from it.
+        key: key of a field in carrier. Returns: first value of the
+        propagation key or an empty list if the key doesn't exist.
+        """
+        raise NotImplementedError()
+
+    def keys(self, carrier: TextMapPropagatorT) -> typing.List[str]:
+        """Function that can retrieve all the keys in a carrier object.
+
+        Args:
+            carrier: and object which contains values that are
+                used to construct a Context. This object
+                must be paired with an appropriate get_from_carrier
+                which understands how to extract a value from it.
+        Returns:
+            list of keys from the carrier.
+        """
+        raise NotImplementedError()
+
+
+class DictGetter(Getter[typing.Dict[str, str]]):
+    def get(
+        self, carrier: typing.Dict[str, str], key: str
+    ) -> typing.List[str]:
+        val = carrier.get(key, None)
```

Comment:
This should check if the value is already a list or collection and return it as is but if the value is a single item, then return it as `[val]`